Unable to train models with generic inner modules

**Describe the bug**
I am writing a model where specific components can be swapped out. To do this i introduce a generic parameter on the module struct and only constrain it on the impl. This works fine until i try training the model. There appears to be an issue with the `Module` derive macro and the `Autodiff` wrapper.

**To Reproduce**
Below is a somewhat minimal reproducer.
```rust
use std::marker::PhantomData;

use burn::backend::{Autodiff, LibTorch};
use burn::backend::libtorch::LibTorchDevice;
use burn::module::{AutodiffModule, Module};
use burn::nn::{Linear, LinearConfig};
use burn::optim::AdamWConfig;
use burn::prelude::Backend;
use burn::tensor::backend::AutodiffBackend;
use burn::train::LearnerBuilder;

#[derive(Module, Debug)]
pub struct ModuleA<B: Backend> {
    layer: Linear<B>
}

#[derive(Module, Debug)]
pub struct ModuleB<B: Backend> {
    layer: Linear<B>
}


#[derive(Module, Debug)]
pub struct ActualModel<B: Backend, T> {
    inner: T,
    _phantom: PhantomData<B>
}

fn train<B: AutodiffBackend>(device: B::Device) {
    let model = ActualModel::<B, _> {
        inner: ModuleA {
            layer: LinearConfig::new(100, 100).init::<B>(&device),

        },
        _phantom: PhantomData,
    };
    let adam = AdamWConfig::new().init::<B, _>();
    let learner = LearnerBuilder::<B, (), (), _, _, _>::new(".out")
        .devices(vec![device])
        .num_epochs(100)
        .build(model, adam, 0.0001);
}

#[test]
fn test_thing() {
    let device = LibTorchDevice::Cpu;

    train::<Autodiff<LibTorch>>(device);
}
```

This yields the following errors:

```
error[E0277]: the trait bound `ModuleA<B>: Module<<B as AutodiffBackend>::InnerBackend>` is not satisfied
  --> src/lib.rs:38:19
   |
38 |     let learner = LearnerBuilder::<B, (), (), _, _, _>::new(".out")
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Module<<B as AutodiffBackend>::InnerBackend>` is not implemented for `ModuleA<B>`, which is required by `ActualModel<B, ModuleA<B>>: AutodiffModule<B>`
   |
note: required for `ActualModel<B, ModuleA<B>>` to implement `AutodiffModule<B>`
  --> src/lib.rs:23:10
   |
23 | #[derive(Module, Debug)]
   |          ^^^^^^ unsatisfied trait bound introduced in this `derive` macro
24 | pub struct ActualModel<B: Backend, T> {
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: required by a bound in `LearnerBuilder::<B, T, V, M, O, S>::new`
  --> /home/rerz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/burn-train-0.13.2/src/learner/builder.rs:66:8
   |
66 |     M: AutodiffModule<B> + core::fmt::Display + 'static,
   |        ^^^^^^^^^^^^^^^^^ required by this bound in `LearnerBuilder::<B, T, V, M, O, S>::new`
...
75 |     pub fn new(directory: &str) -> Self {
   |            --- required by a bound in this associated function
   = note: this error originates in the derive macro `Module` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
   |
29 | fn train<B: AutodiffBackend>(device: B::Device) where ModuleA<B>: Module<<B as AutodiffBackend>::InnerBackend> {
   |                                                 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++


```

```
error[E0277]: the trait bound `ModuleA<B>: Module<<B as AutodiffBackend>::InnerBackend>` is not satisfied
  --> src/lib.rs:38:19
   |
38 |     let learner = LearnerBuilder::<B, (), (), _, _, _>::new(".out")
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Module<<B as AutodiffBackend>::InnerBackend>` is not implemented for `ModuleA<B>`, which is required by `ActualModel<B, ModuleA<B>>: AutodiffModule<B>`
   |
note: required for `ActualModel<B, ModuleA<B>>` to implement `AutodiffModule<B>`
  --> src/lib.rs:23:10
   |
23 | #[derive(Module, Debug)]
   |          ^^^^^^ unsatisfied trait bound introduced in this `derive` macro
24 | pub struct ActualModel<B: Backend, T> {
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: required by a bound in `AdamWConfig::init`
  --> /home/rerz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/burn-core-0.13.2/src/optim/adamw.rs:86:40
   |
86 |     pub fn init<B: AutodiffBackend, M: AutodiffModule<B>>(&self) -> impl Optimizer<M, B> {
   |                                        ^^^^^^^^^^^^^^^^^ required by this bound in `AdamWConfig::init`
   = note: this error originates in the derive macro `Module` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
   |
29 | fn train<B: AutodiffBackend>(device: B::Device) where ModuleA<B>: Module<<B as AutodiffBackend>::InnerBackend> {
   |                                                 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++


```


```
error[E0277]: the trait bound `ModuleA<B>: Module<<B as AutodiffBackend>::InnerBackend>` is not satisfied
  --> src/lib.rs:37:45
   |
37 |     let adam = AdamWConfig::new().init::<B, _>();
   |                                   ----      ^ the trait `Module<<B as AutodiffBackend>::InnerBackend>` is not implemented for `ModuleA<B>`, which is required by `ActualModel<B, ModuleA<B>>: AutodiffModule<B>`
   |                                   |
   |                                   required by a bound introduced by this call
   |
note: required for `ActualModel<B, ModuleA<B>>` to implement `AutodiffModule<B>`
  --> src/lib.rs:23:10
   |
23 | #[derive(Module, Debug)]
   |          ^^^^^^ unsatisfied trait bound introduced in this `derive` macro
24 | pub struct ActualModel<B: Backend, T> {
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
note: required by a bound in `AdamWConfig::init`
  --> /home/rerz/.cargo/registry/src/index.crates.io-6f17d22bba15001f/burn-core-0.13.2/src/optim/adamw.rs:86:40
   |
86 |     pub fn init<B: AutodiffBackend, M: AutodiffModule<B>>(&self) -> impl Optimizer<M, B> {
   |                                        ^^^^^^^^^^^^^^^^^ required by this bound in `AdamWConfig::init`
   = note: this error originates in the derive macro `Module` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
   |
29 | fn train<B: AutodiffBackend>(device: B::Device) where ModuleA<B>: Module<<B as AutodiffBackend>::InnerBackend> {
   |                                                 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++


``` 

The derive macro generates an impl like

```rust
impl<B: Backend> burn::module::Module<B> for ModuleA<B> {
    ...
}
```

But i believe for training another impl like

```rust
impl<B: AutodiffBackend> burn::module::Module<B::InnerBackend> for ModuleA<B> {
    ...
}
``` 

is needed to satisfy the trait bound above.


**Expected behavior**
The code compiles and i am able to train the model.

--- a/crates/burn-derive/src/module/codegen.rs
+++ b/crates/burn-derive/src/module/codegen.rs
@@ -224,11 +224,6 @@ impl GenericsParser {
             );
 
             generics_names_except_backend.extend(quote! { <#ident as burn::module::AutodiffModule<B>>::InnerModule, });
-            module_autodiff.add_predicate(
-                parse_quote! {
-                    #ident: burn::module::Module<B::InnerBackend>
-                }
-            );
 
             module_autodiff.add_predicate(
                 parse_quote! {

--- a/crates/burn-derive/src/module/codegen.rs
+++ b/crates/burn-derive/src/module/codegen.rs
@@ -201,16 +201,9 @@ impl GenericsParser {
 
             module.add_predicate(
                 parse_quote! {
-                    #ident: burn::module::ModuleDisplayDefault
-                }
-            );
-
-            module.add_predicate(
-                parse_quote! {
                     #ident: burn::module::ModuleDisplay
                 }
             );
-
 
             module_autodiff.add_predicate(
                 parse_quote! {
@@ -230,23 +223,10 @@ impl GenericsParser {
                 }
             );
 
-            module_autodiff.add_predicate(
-                parse_quote! {
-                    <#ident as burn::module::AutodiffModule<B>>::InnerModule: burn::module::ModuleDisplay
-                }
-            );
-
-
             generics_names_except_backend.extend(quote! { <#ident as burn::module::AutodiffModule<B>>::InnerModule, });
             module_autodiff.add_predicate(
                 parse_quote! {
                     #ident: burn::module::Module<B::InnerBackend>
-                }
-            );
-
-            module_autodiff.add_predicate(
-                parse_quote! {
-                    #ident: burn::module::ModuleDisplayDefault
                 }
             );
 

